src/mscorlib/shared/System/IO/PathInternal.cs

```diff
@@ -117,6 +117,17 @@ internal static bool AreRootsEqual(string first, string second, StringComparison
         internal static string RemoveRelativeSegments(string path, int skip = 0)
         {
             Debug.Assert(skip >= 0);
+
+            // Moves skip forward to the last character in the segment because for UNC path, the root length doesnot include the last directory separator.
+            // So When we subtract one in GetFullPath it changes the skip length to "\\Server\Shar" instead of "\\Server\Share"
```

Comment:
To me this is a bug. We shouldn't be passing "\\Server\Shar" to this method. You should fix the caller to pass good input, not fix up bad input.